api/src/utils/validate-query.ts

```diff
@@ -11,7 +11,10 @@ const querySchema = Joi.object({
 	group: Joi.array().items(Joi.string()),
 	sort: Joi.array().items(Joi.string()),
 	filter: Joi.object({}).unknown(),
-	limit: Joi.number().integer().min(-1),
+	limit:
+		'MAX_QUERY_LIMIT' in env && env.MAX_QUERY_LIMIT !== -1
+			? Joi.number().integer().min(-1).max(env.MAX_QUERY_LIMIT) // min should be 0
+			: Joi.number().integer().min(-1),
```

Comment:
I think we should also validate the requested limit with the `DEFAULT_QUERY_LIMIT`, so that it errors when the explicitly specified limit within a query exceeds. It's an invalid limit. 🤔 

```suggestion
	limit:
		'MAX_QUERY_LIMIT' in env && env.MAX_QUERY_LIMIT !== -1
			? Joi.number().integer().min(-1).max(env.MAX_QUERY_LIMIT) // min should be 0
			: Joi.number().integer().min(-1).max(env.DEFAULT_QUERY_LIMIT),
```

Reply:
Incorrect interpretation of `DEFAULT_QUERY_LIMIT` being a limit itself, it acts as the limit only when a limit is not provided in a query.